src/internal/noscrypt_logger.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <plog/Log.h>
+#include <noscrypt.h>
+
+/*
+* @brief Logs an error message with the function name and line number where the 
+* error occurred. This is useful for debugging and logging errors in the Noscrypt
+* library.
+*/
+#define NC_LOG_ERROR(result) _printNoscryptError(result, __func__, __LINE__)
```

Comment:
I started thinking this probably shouldn't be an inline function. It's not a lot of code, but I think more than is necessary for an inline function.